src/main/scala/za/co/absa/commons/s3/S3Location.scala

```diff
@@ -49,10 +53,28 @@ object SimpleS3Location {
     }
 
     def isValidS3Path: Boolean = S3LocationRx.pattern.matcher(path).matches
-  }
 
-  def apply(path: String): SimpleS3Location = {
-    path.toSimpleS3Location.getOrElse(throw new IllegalArgumentException(s"Could not parse S3 location from $path!"))
+    def ensureS3LocationEndsWithoutSlash: SimpleS3Location = {
+      val parsedS3Location = apply(path)
+
+      if (parsedS3Location.path.endsWith("/")) parsedS3Location.copy(path=parsedS3Location.path.dropRight(1))
+      else parsedS3Location
+    }
+
+    def ensureS3LocationEndsWithSlash: SimpleS3Location = {
```

Comment:
```suggestion
    def withTrailSlash: SimpleS3Location = {
```
The other function name would respectively change to  `noTrailSlash`

Why:
1. Brevity. Easier to grasp the difference between `withBlah / noBlah` vs. `aaaaaWithBbbbbCcccc / aaaaaWithoutBbbbbCcccc`. Try a few times. I'm struggling with it even here on this line :)

2. The word `ensure` implies action, hence the method, while in fact it's a pure function.

Reply:
I like it, thank you for the review.

The reasoning behind 'ensure' was that it explicitly says that it actually does something (i.e adds or removes '/' according to the situation), but I'm happy to rename it into 'with' prefix :)